### PR TITLE
Microsoft.VCRedist.2015+.x64 version 14.34.31931

### PR DESCRIPTION
**Pull request by TheBossMagnus**
- [ ] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [ ] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/AUTHORING_MANIFESTS.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [ ] Have you tested your manifest locally with `winget install --manifest <path>`?
- [ ] Does your manifest conform to the [1.2 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.2.0)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

-----

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-pkgs/pull/88243)

### CODE REVIEW

File: manifests/m/Microsoft/VCRedist/2015+/x64/14.34.31931/Microsoft.VCRedist.2015+.x64.installer.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# Created using wingetcreate 1.1.2.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.installer.1.2.0.schema.json
+
+PackageIdentifier: Microsoft.VCRedist.2015+.x64
+PackageVersion: 14.34.31931
```
Comment (Trenly) [inline manifests/m/Microsoft/VCRedist/2015+/x64/14.34.31931/Microsoft.VCRedist.2015+.x64.installer.yaml:5]:
```suggestion
PackageVersion: 14.34.31931.0
```

File: manifests/m/Microsoft/VCRedist/2015+/x64/14.34.31931/Microsoft.VCRedist.2015+.x64.installer.yaml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+InstallerSwitches:
+  Silent: /quiet /install
+  SilentWithProgress: /passive /install
+  Upgrade: /norestart
+  Custom: /norestart
+InstallerSuccessCodes:
+- 3010
+UpgradeBehavior: install
+AppsAndFeaturesEntries:
+- DisplayName: Microsoft Visual C++ 2015-2022 Redistributable (x64) - 14.34.31823
```
Comment (Trenly) [inline manifests/m/Microsoft/VCRedist/2015+/x64/14.34.31931/Microsoft.VCRedist.2015+.x64.installer.yaml:23]:
```suggestion
- DisplayName: Microsoft Visual C++ 2015-2022 Redistributable (x64) - 14.34.31931
```

File: manifests/m/Microsoft/VCRedist/2015+/x64/14.34.31931/Microsoft.VCRedist.2015+.x64.installer.yaml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  Publisher: Microsoft Corporation
+  ProductCode: '{ac8ae441-cfc2-41f2-bbca-7b6668740f8d}'
```
Comment (Trenly) [inline manifests/m/Microsoft/VCRedist/2015+/x64/14.34.31931/Microsoft.VCRedist.2015+.x64.installer.yaml:25]:
```suggestion
  ProductCode: '{d4cecf3b-b68f-4995-8840-52ea0fab646e}'

```

File: manifests/m/Microsoft/VCRedist/2015+/x64/14.34.31931/Microsoft.VCRedist.2015+.x64.installer.yaml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+Installers:
+- Architecture: x64
+  InstallerType: burn
+  InstallerUrl: https://download.visualstudio.microsoft.com/download/pr/bcb0cef1-f8cb-4311-8a5c-650a5b694eab/2257B3FBE3C7559DE8B31170155A433FAF5B83829E67C589D5674FF086B868B9/VC_redist.x64.exe
+  InstallerSha256: 2257B3FBE3C7559DE8B31170155A433FAF5B83829E67C589D5674FF086B868B9
+  ProductCode: '{ac8ae441-cfc2-41f2-bbca-7b6668740f8d}'
```
Comment (Trenly) [inline manifests/m/Microsoft/VCRedist/2015+/x64/14.34.31931/Microsoft.VCRedist.2015+.x64.installer.yaml:31]:
```suggestion
  ProductCode: '{d4cecf3b-b68f-4995-8840-52ea0fab646e}'
```

**Review by Trenly — CHANGES_REQUESTED** [review]

File: manifests/m/Microsoft/VCRedist/2015+/x64/14.34.31931/Microsoft.VCRedist.2015+.x64.locale.en-US.yaml [hunk 5]
```diff
@@ -0,0 +1,26 @@
+# Created using wingetcreate 1.1.2.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.defaultLocale.1.2.0.schema.json
+
+PackageIdentifier: Microsoft.VCRedist.2015+.x64
+PackageVersion: 14.34.31931
```
Comment (mdanish-kh) [inline manifests/m/Microsoft/VCRedist/2015+/x64/14.34.31931/Microsoft.VCRedist.2015+.x64.locale.en-US.yaml:5]:
```suggestion
PackageVersion: 14.34.31931.0
```
Must match Control Panel.

File: manifests/m/Microsoft/VCRedist/2015+/x64/14.34.31931/Microsoft.VCRedist.2015+.x64.yaml [hunk 6]
```diff
@@ -0,0 +1,9 @@
+# Created using wingetcreate 1.1.2.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.version.1.2.0.schema.json
+
+PackageIdentifier: Microsoft.VCRedist.2015+.x64
+PackageVersion: 14.34.31931
```
Comment (mdanish-kh) [inline manifests/m/Microsoft/VCRedist/2015+/x64/14.34.31931/Microsoft.VCRedist.2015+.x64.yaml:5]:
```suggestion
PackageVersion: 14.34.31931.0
```
Must match Control Panel.

File: manifests/m/Microsoft/VCRedist/2015+/x64/14.34.31931/Microsoft.VCRedist.2015+.x64.installer.yaml [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+  Upgrade: /norestart
+  Custom: /norestart
+InstallerSuccessCodes:
+- 3010
+UpgradeBehavior: install
+AppsAndFeaturesEntries:
+- DisplayName: Microsoft Visual C++ 2015-2022 Redistributable (x64) - 14.34.31931
+  Publisher: Microsoft Corporation
+  ProductCode: '{d4cecf3b-b68f-4995-8840-52ea0fab646e}'
+
```
Comment (mdanish-kh) [inline manifests/m/Microsoft/VCRedist/2015+/x64/14.34.31931/Microsoft.VCRedist.2015+.x64.installer.yaml:26]:
```suggestion
```
We can get rid of this unnecessary line.

**Review by mdanish-kh — CHANGES_REQUESTED** [review]
After making the below changes, you will also have to change the folder name inside the directory `manifests/m/Microsoft/VCRedist/2015+/64/` from `14.34.31931` to `14.34.31931.0` to match the the PackageVersion. 
A quick and easy way to do this would be to press the <kbd>.</kbd> (dot) key on your keyboard that'll open VSCode Web, where you can find your modified files and modify the folder name from the Explorer menu. After that, you can stage and commit your changes from the Source Control tab (the changes are automatically pushed)

**Comment by TheBossMagnus** [comment]
> After making the below changes, you will also have to change the folder name inside the directory `manifests/m/Microsoft/VCRedist/2015+/64/` from `14.34.31931` to `14.34.31931.0` to match the the PackageVersion.
> A quick and easy way to do this would be to press the . (dot) key on your keyboard that'll open VSCode Web, where you can find your modified files and modify the folder name from the Explorer menu. After that, you can stage and commit your changes from the Source Control tab (the changes are automatically pushed)
> 

I edited the name directly from GitHub... anyway Ty

File: manifests/m/Microsoft/VCRedist/2015+/x64/14.34.31931.0/Microsoft.VCRedist.2015+.x64.locale.en-US.yaml [hunk 8]
```diff
@@ -0,0 +1,26 @@
+# Created using wingetcreate 1.1.2.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.defaultLocale.1.2.0.schema.json
+
+PackageIdentifier: Microsoft.VCRedist.2015+.x64
+PackageVersion: 14.34.31931
```
Comment (mdanish-kh) [inline manifests/m/Microsoft/VCRedist/2015+/x64/14.34.31931.0/Microsoft.VCRedist.2015+.x64.locale.en-US.yaml:5]:
```suggestion
PackageVersion: 14.34.31931.0
```
Missed this one

**Review by mdanish-kh — CHANGES_REQUESTED** [review]

File: manifests/m/Microsoft/VCRedist/2015+/x64/14.34.31931.0/Microsoft.VCRedist.2015+.x64.installer.yaml [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+InstallerSwitches:
+  Silent: /quiet /install
+  SilentWithProgress: /passive /install
+  Upgrade: /norestart
+  Custom: /norestart
+InstallerSuccessCodes:
+- 3010
+UpgradeBehavior: install
+AppsAndFeaturesEntries:
+- DisplayName: Microsoft Visual C++ 2015-2022 Redistributable (x64) - 14.34.31931.0
```
Comment (mdanish-kh) [inline manifests/m/Microsoft/VCRedist/2015+/x64/14.34.31931.0/Microsoft.VCRedist.2015+.x64.installer.yaml:23]:
```suggestion
- DisplayName: Microsoft Visual C++ 2015-2022 Redistributable (x64) - 14.34.31931
```
Must match Control Panel
image

**Review by mdanish-kh — CHANGES_REQUESTED** [review]

**Review by mdanish-kh — APPROVED** [review]

**Pull request merged**